src/sst/core/baseComponent.cc

```diff
@@ -490,10 +524,16 @@ BaseComponent::getSimulationOutput() const
     return Simulation_impl::getSimulationOutput();
 }
 
+SimTime_t
+BaseComponent::getCurrentSimTime(TimeConverter& tc) const
+{
+    return tc.convertFromCoreTime(sim_->getCurrentSimCycle());
+}
+
 SimTime_t
```

Comment:
May want to move this definition to the header file so it can be inlined.

Reply:
I did not make this change because it would require `sim_` to be accessible in the header.